src/unix_mount_entry.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2013-2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib::translate::*;
+use glib::GString;
+use std::cmp;
+use std::mem;
+use Icon;
+
+glib_wrapper! {
+    #[derive(Debug, Hash)]
+    pub struct UnixMountEntry(Boxed<gio_sys::GUnixMountEntry>);
+
+    match fn {
+        copy => |ptr| gobject_sys::g_boxed_copy(gio_sys::g_unix_mount_entry_get_type(), ptr as *mut _) as *mut gio_sys::GUnixMountEntry,
+        free => |ptr| gobject_sys::g_boxed_free(gio_sys::g_unix_mount_entry_get_type(), ptr as *mut _),
+        get_type => || gio_sys::g_unix_mount_entry_get_type(),
+    }
+}
+
+impl UnixMountEntry {
+    pub fn new_at<P: AsRef<std::path::Path>>(mount_path: P) -> (UnixMountEntry, u64) {
+        unsafe {
+            let mut time_read = mem::MaybeUninit::uninit();
+            let ret = from_glib_full(gio_sys::g_unix_mount_at(
+                mount_path.as_ref().to_glib_none().0,
+                time_read.as_mut_ptr(),
+            ));
+            let time_read = time_read.assume_init();
+            (ret, time_read)
+        }
+    }
+
+    #[cfg(any(feature = "v2_52", feature = "dox"))]
+    pub fn new_for<P: AsRef<std::path::Path>>(file_path: P) -> (UnixMountEntry, u64) {
+        unsafe {
+            let mut time_read = mem::MaybeUninit::uninit();
+            let ret = from_glib_full(gio_sys::g_unix_mount_for(
+                file_path.as_ref().to_glib_none().0,
+                time_read.as_mut_ptr(),
+            ));
+            let time_read = time_read.assume_init();
+            (ret, time_read)
+        }
+    }
+
+    pub fn get() -> (Vec<UnixMountEntry>, u64) {
```

Comment:
Maybe this should be `get_mounts()`, i.e. plural? It's not getting a single one but all of them

Reply:
Or `get_mount_entries()`

Reply:
agree